inc/entity.class.php

```diff
@@ -71,6 +71,74 @@ public function post_purgeItem() {
       ]);
    }
 
+   public function calculateGlobalDefault() {
+      if (!isset($_SESSION['credit']['post_update'])) {
+         if ($this->fields['is_default_followup']
+            && $this->fields['is_default_task']
+            && $this->fields['is_default_solution']) {
+               $this->fields['is_default'] = 1;
+         } else if (!$this->fields['is_default_followup']
+            && !$this->fields['is_default_task']
+            && !$this->fields['is_default_solution']) {
+            !$this->fields['is_default'] = 0;
+         } else {
+            $this->fields['is_default'] = 2;
+         }
+         $_SESSION['credit']['post_update'] = true;
+         $this->update($this->fields);
+      } else {
+         unset($_SESSION['credit']['post_update']);
+      }
+   }
+
+   public function post_updateItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
+
+   public function post_addItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
+
+   public function prepareInputForAdd($input) {
+
+      $credit =new PluginCreditEntity();
+      $message = "";
+
+      if ($input['is_default_followup']) {
+         if (count($credit->find([
+            "entities_id" => $input["entities_id"],
+            "is_default_followup"  => $input['is_default_followup']
+         ])) > 0) {
+            $message .= __("Default credit for followups already exist", "credit")."</br>";
+         }
+      }
```

Comment:
1. Checking `array_key_exists('is_default_followup', $input)` could prevent `undefined index` errors when add/update is done by API.
2. Directly using `1` in criteria seems simplier.
3. I think we should filter on active vouchers only.

```suggestion
      if (array_key_exists('is_default_followup', $input) && (int)$input['is_default_followup'] === 1) {
         $existing_defaults = $credit->find(
            [
               "entities_id" => $input["entities_id"],
               "is_default_followup"  => 1,
               "is_active" => 1
            ]
         );
         if (count($existing_defaults) > 0) {
            $message .= __("Default credit for followups already exist", "credit")."</br>";
         }
      }
```